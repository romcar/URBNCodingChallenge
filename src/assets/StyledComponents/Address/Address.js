import styled from 'styled-components';
import colors from '../../theme';

export default styled.div`
  grid-row-start: 7;
  grid-row-end: 9;
  grid-column-start: 10;
  grid-column-end: 16;
  color: ${colors.eventLocation};
  margin-top: 15px;

  a {
    text-decoration: none;
    color: ${colors.eventLocation};

    :hover {
      color: ${colors["link--hover"]}
    }
  }
`;
