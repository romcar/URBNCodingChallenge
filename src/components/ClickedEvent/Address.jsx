import React from 'react'
import StyledAddress from '../../assets/StyledComponents/Address/Address';
export default ({ venue_name, venue_address, city_name, region_abbr, postal_code, venue_url }) => {
  return (
    <StyledAddress className="event-modal-address">
      <a href={venue_url} className="venue-name">{venue_name}</a>
      <div className="venue-address">
        {`${venue_address} ${city_name}, ${region_abbr}. ${postal_code}`}
      </div>
    </StyledAddress>
  )
}
