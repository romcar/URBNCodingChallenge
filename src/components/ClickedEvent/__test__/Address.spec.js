import Adapter from 'enzyme-adapter-react-16';
import Address from '../Address';
import { configure, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import testData from '../../../../fakeSeed';
import React from 'react';

configure({ adapter: new Adapter() });

let testEvent = null;
let wrapper = null;

describe('Address Component', () => {
  beforeEach(() => {
    testEvent = testData[0];
    wrapper = shallow(<Address {...testEvent} />);

  });

  it('Should render component', () => {
    expect(wrapper.find('.event-modal-address').length).toEqual(1);
  });

  it('Should display the correct address for the event', () => {
    const expected = `${testEvent.venue_address} ${testEvent.city_name}, ${testEvent.region_abbr}. ${testEvent.postal_code}`
    expect(wrapper.find('.venue-address').render().text()).toEqual(expected);
  });

  it('Should display the correct venue for the event', () => {
    const expected = testEvent.venue_name;
    expect(wrapper.find('.venue-name').render().text()).toEqual(expected);
  });

  it('Should have the correct url attached to the venue name', () => {
    expect(wrapper.find('.venue-name').filterWhere(item => {
      return item.render()[0].attribs.href === testEvent.venue_url;
    }).length).toEqual(1);
  });

  it('Should match the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});