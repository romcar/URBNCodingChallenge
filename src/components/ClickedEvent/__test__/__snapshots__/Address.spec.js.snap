// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Component Should match the snapshot 1`] = `
<styled.div
  className="event-modal-address"
>
  <a
    className="venue-name"
    href="http://philadelphia.eventful.com/venues/mann-center-for-the-performing-arts-/V0-001-000282568-8?utm_source=apis&utm_medium=apim&utm_campaign=apic"
  >
    Mann Center for the Performing Arts
  </a>
  <div
    className="venue-address"
  >
    5201 Parkside Avenue Philadelphia, PA. 19131
  </div>
</styled.div>
`;
